### PR TITLE
Register-tabcompletion function for repository param

### LINKED ISSUE #312: -Repository parameter should have tab completion
The -Repository parameter in Install-Module (and other cmdlets) should have tab completion over the known repositories. This should be easy because you can easily get repository names by running Get-PSRepository.

### PR DESCRIPTION
**Pull request by edyoung**
Resolves #312

### CODE REVIEW

**Comment by edyoung** [comment]
Looks like Register-ArgumentCompleter is not supported in PowerShell older than 5.0. I'll add a try/catch and a test for existence of the cmdlet. #Closed

**Comment by bmanikm** [comment]
I am fine with conditionally enabling this functionality.

---
In reply to: [426366957](https://github.com/PowerShell/PowerShellGet/pull/339#issuecomment-426366957) [](ancestors = 426366957)

File: PowerShellGet/private/modulefile/PartTwo.ps1 [hunk 1]
```diff
@@ -26,6 +26,29 @@ if(-not (Microsoft.PowerShell.Management\Test-Path -Path $script:MyDocumentsInst
...
+    "Save-Module"
+    "Save-Script")
+
+Register-ArgumentCompleter -CommandName $commandsWithRepositoryParameter -ParameterName Repository -ScriptBlock {
+    param($commandName, $parameterName, $wordToComplete, $commandAst, $fakeBoundParameter) 
+    
+    Get-PSRepository -Name "$wordTocomplete*"-ErrorAction SilentlyContinue -WarningAction SilentlyContinue | Foreach-Object { 
+        [System.Management.Automation.CompletionResult]::new($_.Name, $_.Name, 'ParameterValue', $_.Name) 
+    } 
+}
```
Comment (bmanikm) [inline PowerShellGet/private/modulefile/PartTwo.ps1]: Please consider adding this for the Name param of below cmdlets.

Get-PSRepository
Set-PSRepository
Unregister-PSRepository
 #Closed

File: PowerShellGet/private/modulefile/PartTwo.ps1 [hunk 2]
```diff
@@ -26,6 +26,57 @@ if(-not (Microsoft.PowerShell.Management\Test-Path -Path $script:MyDocumentsInst
                                                      -WhatIf:$false
 }
 
+function Add-ArgumentCompleter()
```
Comment (bmanikm) [inline PowerShellGet/private/modulefile/PartTwo.ps1]: Please move this utility function into a separate file under ../../private/functions/ folder. #Closed

File: PowerShellGet/private/modulefile/PartTwo.ps1 [hunk 3]
```diff
@@ -26,6 +26,57 @@ if(-not (Microsoft.PowerShell.Management\Test-Path -Path $script:MyDocumentsInst
...
+{
+    [CmdletBinding()]
+    param(
+        [string[]]$cmdlets,
+        [string]$parameterName
+    )
```
Comment (bmanikm) [inline PowerShellGet/private/modulefile/PartTwo.ps1]: Please consider adding the mandatory parameter attribute for this two parameters. #Closed

**Review by bmanikm — APPROVED** [review]
:shipit:

**Pull request merged**